mlos_bench/mlos_bench/services/remote/azure/azure_flex.py

```diff
@@ -0,0 +1,207 @@
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+"""
+A collection Service functions for managing VMs on Azure.
+"""
+import logging
+
+from typing import Any, Dict, Optional, Tuple
+
+import requests
+
+from mlos_bench.environments.status import Status
+from mlos_bench.services.base_service import Service
+from mlos_bench.services.types.authenticator_type import SupportsAuth
+from mlos_bench.services.types.remote_config_type import SupportsRemoteConfig
+from mlos_bench.util import check_required_params, merge_parameters
+
+_LOG = logging.getLogger(__name__)
+
+
+class AzureFlexConfigService(Service, SupportsRemoteConfig):
+    """
+    Helper methods to configure Azure Flex services.
+    """
+
+    _REQUEST_TIMEOUT = 5  # seconds
+
+    # Azure Flex Services Configuration REST API as described in
+    # https://learn.microsoft.com/en-us/rest/api/mysql/flexibleserver/configurations
+
+    _URL_CONFIGURE = (
+        "https://management.azure.com" +
+        "/subscriptions/{subscription}" +
+        "/resourceGroups/{resource_group}" +
+        "/providers/{provider}" +
+        "/{server_type}/{vm_name}" +
+        "/{update}" +
+        "?api-version={api_version}"
+    )
+
+    def __init__(self,
+                 config: Optional[Dict[str, Any]] = None,
+                 global_config: Optional[Dict[str, Any]] = None,
+                 parent: Optional[Service] = None):
+        """
+        Create a new instance of Azure services proxy.
+
+        Parameters
+        ----------
+        config : dict
+            Free-format dictionary that contains the benchmark environment
+            configuration.
+        global_config : dict
+            Free-format dictionary of global parameters.
+        parent : Service
+            Parent service that can provide mixin functions.
+        """
+        super().__init__(config, global_config, parent)
+
+        check_required_params(self.config, {
+            "subscription",
+            "resourceGroup",
+            "provider",
+        })
+
+        (self._is_batch, is_flex, api_version) = {
+            "Microsoft.DBforMySQL": (True, True, "2022-01-01"),
+            "Microsoft.DBforMariaDB": (False, False, "2018-06-01"),
+            "Microsoft.DBforPostgreSQL": (False, True, "2022-12-01"),
+        }[self.config["provider"]]
```

Comment:
This makes me a little nervous - seems like a support headache.  Is there a more generic way we can do more of this with just configs?

Reply:
yes, we can move all these parameters to the config - in fact, I had it this way a few commits back. Then, say, a `[service-flex-mysql.jsonc](https://github.com/microsoft/MLOS/pull/520/files#diff-c2b0c5fd6d3c817c136510b21212d9b22212b0a9254228f7593bb0fe460b2c72)` would look like this:
```javascript
{
    "class": "mlos_bench.services.remote.azure.AzureFlexConfigService",

    "config": {

        "subscription": "PLACEHOLDER; AZURE SUBSCRIPTION ID",
        "resourceGroup": "PLACEHOLDER; e.g., os-autotune",
        "vmName": "PLACEHOLDER; e.g., mlos-vm-mysql",

        "provider": "Microsoft.DBforMySQL",
        "supportsBatchUpdate": true,
        "isFlex": true,
        "apiVersion": "2022-01-01",

        "requestTimeout": 20,   // HTTP request timeout in seconds
        "pollInterval": 20,     // Poll every 20 seconds
        "pollTimeout": 600      // Give up polling after 10 minutes
    }
}
```
I am fine with either way

Reply:
The config only way is less friendly, but more easily fixable and expandable overtime with fewer interactions from us maintainers, so I think it might be the better way to go.
One thing we could do is bake in a few "currently known sane defaults" when some of the config values aren't provided.
Either way this would need to be well documented.

Reply:
Also, can the Service be just `remote.azure.AzureSaaSService` instead of specifically `Flex` services?  Or is that too difficult?

Reply:
no, I think it makes sense. Not all DBs on Azure are Flex servers, anyway

Reply:
also, just pushed a new version of the service and configs